mmv1/third_party/terraform/resources/resource_bigtable_gc_policy.go

```diff
@@ -288,13 +320,22 @@ func generateBigtableGCPolicy(d *schema.ResourceData) (bigtable.GCPolicy, error)
 	mode := d.Get("mode").(string)
 	ma, aok := d.GetOk("max_age")
 	mv, vok := d.GetOk("max_version")
+	gcRules, gok := d.GetOk("gc_rules")
 
-	if !aok && !vok {
+	if !aok && !vok && !gok {
 		return bigtable.NoGcPolicy(), nil
 	}
 
 	if mode == "" && aok && vok {
-		return nil, fmt.Errorf("If multiple policies are set, mode can't be empty")
+		return nil, fmt.Errorf("if multiple policies are set, mode can't be empty")
+	}
+
+	if gok {
```

Comment:
This means gc_rules takes precedent over "mode", "max_age" and "max_version". Can we add this in the comment instead of just "Conflicts with "mode", "max_age" and "max_version"."?

Reply:
Conflicting with others means they can't both exist, I guess we don't have to mention which one take precedent. However we should try to discourage users to use other fields ("mode", "max_age" and "max_version") and even mark them as deprecated. I there a way to do that? What is the deprecation policy?

Reply:
I don't think deprecating the existing fields is necessary  for now, since there are simpler use case that doesn't require nested policy, for example, if customers only need `age() > 2h`.

Reply:
I believe something has to be done to "mode", "max_version" and "max_age", but the change doesn't need to be in this PR. We can chat offline.

Reply:
Synced offline. Decided to keep other fields as deprecation in another PR.